### DISCUSSION TITLE
Create Disc-O vehicle using roto-drop/observation tower

### DISCUSSION THREAD

**Discussion by astraylife1** [OpenRCT2/OpenRCT2#19429] [Feature requests]
If the observation tower ride vehicle is used on a disc-o track, the vehicle will spin automatically. I don't have the technical abilities but it seems like it would be easy to copy the sprites of the roto-drop vehicle onto a version of the observation wheel vehicle to create a spinning vehicle for disc-o type rides. Maybe some work needs to be done so that the vehicle doesn't bleed through the track. 

Anyway, I was just playing around and noticed I can have a spinning disc-o ride with the observation wheel vehicles. Just wish the roto-drop vehicle would spin automatically too.

**Comment by karst** [comment]
I believe @spacek531 was working on a Disk-O and Extreme Spinner. Not sure if there's any progress on that though. If we want to include those we will have to contact Zamperla & Mack. The latter seems to be pretty active with the coaster community so higher chances for that.

**Reply by astraylife1** [reply]
Yeah, I'm not sure how it all works. I figured you could just copy the sprites from the roto drop on to the observation wheel vehicle. Maybe it is more complicated than that.

Also, why would we need to contact ride manufacturers? Just call the vehicle something generic. There are several companies that produce disc-o type attractions.

**Comment by spacek531** [comment]
The code that makes roto-drop spin and the code that makes observation tower spin are very different. The code that makes disko spin may be related to one of those two, but may be completely different as well.

For an official disko object, I would make a vehicle that only resembles the layout of a Zamperla Disk'O car. Zamperla Disk'O vehicles have 24 or 40 seats but RCT2 has a limit of 32 seats. They also have only one opening because the stations are always set up the same, but mine will have one on either side for operational flexibility. The drawing engine limits vehicle sizes to only slightly larger than a Zamperla Disk'O-sized track -- a Zamperla Disk'O vehicle is much larger than its track and will clip hard.

**Reply by astraylife1** [reply]
I always use the hybrid track with the roto drop vehicles. Gets decent enough stats for a family flat ride. Just wish the roto cars would spin. But it seems it’s not an easy task
